buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java

```diff
@@ -408,7 +407,17 @@ public int numDirectArenas() {
      * Return the number of thread local caches used by this {@link PooledByteBufAllocator}.
      */
     public int numThreadLocalCaches() {
-        return threadCache.caches.get();
+        PoolArena<?>[] arenas = heapArenas != null ? heapArenas : directArenas;
```

Comment:
@buchgr shouldnt you take both (heap and direct arenas) into account ? Each of them may be null or just one of them, or even none.